sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -812,150 +812,234 @@ class HiveDDLSuite
     }
   }
 
-  test("CREATE TABLE LIKE a temporary view") {
-    val sourceViewName = "tab1"
-    val targetTabName = "tab2"
-    withTempView(sourceViewName) {
-      withTable(targetTabName) {
-        spark.range(10).select('id as 'a, 'id as 'b, 'id as 'c, 'id as 'd)
-          .createTempView(sourceViewName)
-        sql(s"CREATE TABLE $targetTabName LIKE $sourceViewName")
-
-        val sourceTable = spark.sessionState.catalog.getTempViewOrPermanentTableMetadata(
-          TableIdentifier(sourceViewName))
-        val targetTable = spark.sessionState.catalog.getTableMetadata(
-          TableIdentifier(targetTabName, Some("default")))
-
-        checkCreateTableLike(sourceTable, targetTable)
+  test("CREATE TABLE LIKE a temporary view [LOCATION]...") {
+    var createdTableType = "MANAGED"
+    for ( i <- 0 to 1 ) {
```

Comment:
I write this for the purpose of reusing this piece of  public code, because the basic logic of these two scenarios are almost the same.